zebra-rpc/src/methods/tests/snapshot.rs

```diff
@@ -183,37 +184,48 @@ fn snapshot_rpc_getblockchaininfo(info: GetBlockChainInfo, settings: &insta::Set
     });
 }
 
-/// Snapshot `getaddressbalance` response, using `cargo insta` and RON serialization.
+/// Snapshot `getaddressbalance` response, using `cargo insta` and JSON serialization.
 fn snapshot_rpc_getaddressbalance(address_balance: AddressBalance, settings: &insta::Settings) {
     settings.bind(|| insta::assert_json_snapshot!("get_address_balance", address_balance));
 }
 
-/// Snapshot `getblock` response, using `cargo insta` and RON serialization.
-fn snapshot_rpc_getblock(block: GetBlock, settings: &insta::Settings) {
-    settings.bind(|| insta::assert_json_snapshot!("get_block", block));
+/// Check `getblock` response, using `cargo insta`, JSON serialization, and block test vectors.
+///
+/// The snapshot file does not contain any data, but it does enforce the response format.
+fn snapshot_rpc_getblock(block: GetBlock, block_data: &[u8], settings: &insta::Settings) {
+    let block_data = hex::encode(block_data);
+
+    settings.bind(||
+                  insta::assert_json_snapshot!("get_block", block, { "." => insta::dynamic_redaction(move |value, _path| {
+                      // assert that the block data matches, without creating a 1.5 kB snapshot file
+                      assert_eq!(value.as_str().unwrap(), block_data);
+
+                      "[block_data]"
```

Comment:
```suggestion
                      "[BlockData]"
```

I will like to be consistent with other redactions like `Height` or `SubVersion`. We might change thoses or this one to lower case.